From 76d59839a8f1afbcc5f96f629e66ac350a3abf0d Mon Sep 17 00:00:00 2001
From: Steve Wainstead <wainstead@gmail.com>
Date: Sat, 4 Dec 2010 00:55:47 +0000
Subject: [PATCH 075/223] Cleaning up warnings generated when compiled with
 -pendantic: comparing signed and unsigned ints. The
 signed ints were just temporary variables used as
 loop counters.

---
 server/code_gen.c |    7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/server/code_gen.c b/server/code_gen.c
index ef96d88..583bdbc 100644
--- a/server/code_gen.c
+++ b/server/code_gen.c
@@ -196,7 +196,7 @@ emit_extended_byte(Byte b, State * state)
 static int
 add_known_fixup(Fixup f, State * state)
 {
-    int i;
+    unsigned int i;
 
     if (state->num_fixups == state->max_fixups) {
 	unsigned new_max = 2 * state->max_fixups;
@@ -448,7 +448,7 @@ static void
 enter_loop(int id, Fixup top_label, unsigned top_stack,
 	   int bottom_label, unsigned bottom_stack, State * state)
 {
-    int i;
+    unsigned int i;
     Loop *loop;
 
     if (state->num_loops == state->max_loops) {
@@ -1134,7 +1134,8 @@ stmt_to_code(Stmt * stmt, GState * gstate)
 {
     State state;
     Bytecodes bc;
-    int old_i, new_i, fix_i;
+    unsigned int old_i, new_i, fix_i;
+
 #ifdef BYTECODE_REDUCE_REF
     int *bbd, n_bbd;		/* basic block delimiters */
     unsigned varbits;		/* variables we've seen */
-- 
1.7.9.5

